astyanax-queue/src/main/java/com/netflix/astyanax/recipes/queue/MessageConsumerImpl.java

```diff
@@ -46,15 +46,18 @@ public MessageConsumerImpl(ShardedDistributedMessageQueue q) {
         // TODO: Read full itemsToPop instead of just stopping when we get the first successful set
         List<MessageContext> messages = null;
         while (true) {
+            boolean success = false;
             MessageQueueShard partition = queue.shardReaderPolicy.nextShard();
             if (partition != null) {
                 try {
                     messages = readAndReturnShard(partition, itemsToPop);
+                    success = true;
                     if (messages != null && !messages.isEmpty()) {
                         return messages;
                     }
                 } finally {
-                    queue.shardReaderPolicy.releaseShard(partition, messages == null ? 0 : messages.size());
```

Comment:
I thought about:
                } catch (MessageQueueException e) {
                    queue.shardReaderPolicy.releaseShard(partition, -1);
                    throw e;
                } catch (BusyLockException e) {
                    queue.shardReaderPolicy.releaseShard(partition, -1);
                    throw e;
                } catch (InterruptedException e ) {
                    queue.shardReaderPolicy.releaseShard(partition, -1);
                    throw e;
                } catch (RuntimeException e) {
                    queue.shardReaderPolicy.releaseShard(partition, -1);
                    throw e;
                }
ands that approach seemed even more unwieldy and it misses any exceptions that may be added in the future.  Capturing Throwable cause a signature change :(